### PR TITLE
181 create modals for upgrades app

### PR DESCRIPTION
**Pull request by MangoMagoCane**
Added an upgrade modal as well as getting carried away and adding type to the `styleDataSlice`

image
image

### CODE REVIEW

**Review by JudeKlatte — APPROVED** [review]
ヾ(⌐■_■)ノ♪

File: frontend/src/components/UpgradeModal.tsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+      style={{
+        backgroundColor: "orange",
+        borderRadius: "6px",
+        position: "fixed",
+        display: "flex",
+        flexDirection: "column",
+        alignItems: "center",
+        bottom: document.defaultView.innerHeight - up.yPos,
+        left: up.xPos,
+        // width: width,
```
Comment (jvgusch) [inline frontend/src/components/UpgradeModal.tsx:45]: I'm assuming you meant to remove this?

File: frontend/src/slices/styleDataSlice.ts [hunk 2]
```diff
@@ -14,7 +23,8 @@ export const validStyleFunctions = {
     return check.color === strColor.toLowerCase();
   },
   textFont: (fontType) => {
-    return fontType;
+    // return fontType;
```
Comment (jvgusch) [inline frontend/src/slices/styleDataSlice.ts:26]: This can be removed.

**Review by jvgusch — APPROVED** [review]

File: frontend/src/slices/styleDataSlice.ts (hunk 2 above)
Comment (MangoMagoCane) [reply frontend/src/slices/styleDataSlice.ts]: the comment was originally documentation for what was previously returned but I made that clearer now.

File: frontend/src/components/UpgradeModal.tsx (hunk 1 above)
Comment (MangoMagoCane) [reply frontend/src/components/UpgradeModal.tsx]: yep.

**Review by kavyab2006 — CHANGES_REQUESTED** [review]
Make it so the modals close when you click the upgrade again

**Comment by MangoMagoCane** [comment]
> Make it so the modals close when you click the upgrade again

Added!

**Review by kavyab2006 — APPROVED** [review]
Do we want the currency to be called pixels or pixelpayout  or something else? I am okay with anything, but I think it should be consistent because in the modals, the currency is called pixels, whereas in the taskbar, it is called pixelpayout. Picking one might be better. But it's your choice. 

Also, did you want the progress bar to change colors as they got closer to buying the upgrade, because the progress bar currently stays blue the entire time. If you had time, I think it would be a cool feature. If you decided it was better not to add it, that's fine too.

Overall, I really like it!

**Review by JudeKlatte — APPROVED** [review]
nice 👍

**Pull request merged**